core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -146,7 +146,7 @@ public Set<Integer> struct(Types.StructType struct, Iterable<Set<Integer>> field
 
             // visit children to add more ids
             Iterator<Set<Integer>> iter = fieldResults.iterator();
-            while (shouldContinue() && iter.hasNext()) {
```

Comment:
why is the `shouldContinue` removed?

Reply:
I am trying to address the comment from @rdblue from https://github.com/apache/iceberg/pull/13039#discussion_r2254733603. I think he is suggesting it's ok to visit inside the nested field instead of checking before visit.